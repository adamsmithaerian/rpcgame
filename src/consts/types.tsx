export enum Shape {
  Rock,
  Paper,
  Scissors
}

export enum Result {
  Win,
  Lose,
  Tie
}

export const shapeNames: Array<String> = ['Rock', 'Paper', 'Scissors'];

export const resultNames: Array<String> = ['Win', 'Lose', 'Tie'];
