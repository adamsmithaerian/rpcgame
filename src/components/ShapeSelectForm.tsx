import * as React from 'react';
import ShapeSelect from './ShapeSelect';
import { Shape } from '../consts/types';

interface ShapeSelectProps {
  onSubmitShape: () => void;
  onSelectShape: (e: any) => void;
  selectedShape: Shape;
}

class ShapeSelectForm extends React.Component<ShapeSelectProps, any> {
  render() {
    return (
      <form onSubmit={this.props.onSubmitShape}>
        <label className="text">Please select your shape</label>

        <ShapeSelect
          onChange={this.props.onSelectShape}
          value={this.props.selectedShape}
        />
        <input type="submit" value="Submit" />
      </form>
    );
  }
}

export default ShapeSelectForm;
