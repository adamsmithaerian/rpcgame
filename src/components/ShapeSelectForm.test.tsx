import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';

import ShapeSelectForm from './ShapeSelectForm';
import { Shape } from '../consts/types';

describe('ShapeSelectForm component should', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(
      <ShapeSelectForm
        onSubmitShape={() => {}}
        onSelectShape={() => {}}
        selectedShape={Shape.Rock}
      />,
      div
    );
  });

  it('match snapshot', () => {
    const tree = renderer
      .create(
        <ShapeSelectForm
          onSubmitShape={() => {}}
          onSelectShape={() => {}}
          selectedShape={Shape.Rock}
        />
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  });   
});
