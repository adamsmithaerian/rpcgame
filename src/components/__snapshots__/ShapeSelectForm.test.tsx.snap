// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShapeSelectForm component should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    className="text"
  >
    Please select your shape
  </label>
  <div
    className="Shape-select"
  >
    <select
      onChange={[Function]}
      value={0}
    >
      <option
        value={0}
      >
        Rock
      </option>
      <option
        value={1}
      >
        Paper
      </option>
      <option
        value={2}
      >
        Scissors
      </option>
    </select>
  </div>
  <input
    type="submit"
    value="Submit"
  />
</form>
`;
